src/cli.ts

```diff
@@ -66,6 +66,10 @@ export class CliImpl implements Cli {
       let error: string | Error;
       tkn.stdout.on('data', (data) => {
         stdout += data;
+        const regexStatus = /\sSucceeded\s|\sFailed\s/;
+        if(regexStatus.test(data.toString())) {
+          resolve({ error, stdout });
+        }
```

Comment:
We need to separate API and implementation for executing CLI commands and CLI commands with watch feature. 
I suggest to create another method/function for executing CLI command with watch, it should has event system, when new output appears, and ability to stop/cancel execution. Caller of this API should be able to provide an event listener for output, and should have ability to terminate/stop running process when there are no sense to run it(pipeline execution finished, user change view, etc.).